src/values/acceleration-structure.cxx

```diff
@@ -0,0 +1,2224 @@
+/* © SPIRV-Interpreter @ https://github.com/mmoult/SPIRV-Interpreter
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+module;
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <stack>
+#include <string>
+#include <tuple>
+
+// TODO: plan to remove/change header(s) below
+#include <iostream>
+
+// TODO: probably want to reduce GLM files to link
+#include <glm/ext.hpp>
+
+#include "../external/spirv.hpp"
+#include "type.hpp"
+#include "value.hpp"
+
+export module value.accelerationStructure;
+import value.aggregate;
+import value.primitive;
+
+namespace Util {
+
+/// @brief Get a repeated string.
+/// @param num number of times to repeat a string.
+/// @param str string to repeat.
+/// @return single string of the repeated string.
+std::string repeatedString(const unsigned num, const std::string& str) {
```

Comment:
We could be more clever about this since we know ahead of time how long the result will be (and therefore, we should be able to reserve that exact space), but I suppose this is all debug-only, so it shouldn't really matter.